src/box/lua/net_box.c

```diff
@@ -2313,6 +2394,62 @@ netbox_transport_on_event(struct netbox_transport *transport,
 	lua_call(L, watch.data != NULL ? 3 : 2, 0);
 }
 
+/**
+ * Write response to buffer.
+ */
+static inline void
+netbox_write_response_to_buffer(const char *data, const char *data_end,
+				struct lua_State *L, struct ibuf *buffer,
+				bool skip_header)
+{
+	/* Copy xrow.body to user-provided buffer. */
+	size_t data_len = data_end - data;
+	bool return_table = false;
+	if (skip_header) {
+		assert(mp_typeof(*data) == MP_MAP);
+		uint32_t map_size = mp_decode_map(&data);
+		uint32_t key = mp_decode_uint(&data);
+		assert(key == IPROTO_DATA);
+		(void)key;
+		if (map_size > 1) {
+			/*
+			 * The map has more than one element iff it
+			 * contains IPROTO_DATA and IPROTO_POSITION
+			 * and they are placed exactly in this order.
+			 */
+			assert(map_size == 2);
+			/* Find the end of IPROTO_DATA and its len. */
+			const char *iproto_position = data;
+			mp_next(&iproto_position);
+			data_len = iproto_position - data;
+			/* Create table to return 2 values. */
+			return_table = true;
+			lua_createtable(L, 2, 0);
+			/* Skip IPROTO_POSITION key. */
+			assert(*iproto_position == IPROTO_POSITION);
```

Comment:
Nit: Please don't implicitly assume that `mp_decode_uint(x) == x` in this case.

Reply:
Done